popup.js

```diff
@@ -83,11 +83,47 @@ document.addEventListener('DOMContentLoaded', function () { // this function  st
                 console.log("user has deleted the board manually.");
                 board_create_function();
             });
-
+        
+        // Populating Fields
         chrome.tabs.getSelected(null, function (tab) {
             document.getElementById('data_url').value = tab.url;
         });
 
+        function getFieldsFromDOM() {
+            var fields = []
+            //Account for Companies which have no URL/Link
+            try {
+                var url = document.getElementsByClassName("jobs-details-top-card__company-url")[0].innerText.trim();
+            } catch {
+            }
+            if (url) {
+                fields.push(document.getElementsByClassName("jobs-details-top-card__company-url")[0].innerText.trim())
```

Comment:
not sure if this would work, but maybe you can remove the if-else statement altogether and put the if content in try and else in catch. I might be misunderstanding this but just a thought!

Reply:
This might break something so let me know if this is the best way! Just trying to see any optimizations.

Reply:
Sweet, I'll give it a go and get back to you